src/AppInstallerCLICore/Workflows/UpdateFlow.cpp

```diff
@@ -86,16 +89,25 @@ namespace AppInstaller::CLI::Workflow
                 context.Add<Execution::Data::PackageVersion>(std::move(packageVersion));
                 context.Add<Execution::Data::Installer>(std::move(installer));
 
-                updateFound = true;
-                break;
+                    updateFound = true;
+                    break;
+                }
+                else
+                {
+                    // Any following versions are not applicable
+                    break;
+                }
             }
-            else
+        }
+        else
+        {
+            // the package has a unknown version and the user did not request to upgrade it anyway.
+            if (m_reportUpdateNotFound) 
             {
-                // Any following versions are not applicable
-                break;
+                context.Reporter.Info() << Resource::String::UpdateNotApplicable << std::endl;

```

Comment:
I think this message could be more specific so a user knows that they can still update if they add the flag.

Reply:
In the latest commit: 
```
PS C:\Users\easton\Downloads> wingetdev upgrade GOG.Galaxy
This package's version number cannot be determined. To upgrade it anyway, add the argument --include-unknown to your previous command.
PS C:\Users\easton\Downloads>
```